temporaldata/temporaldata.py

```diff
@@ -2287,15 +2287,22 @@ def __or__(self, other):
                 else:
                     assert current_start is not None
                     if not end_still_coming:
-                        # we have an opening and a closing paranthesis
-                        if current_start != current_end:
-                            # we have a non-zero interval
-                            start = np.append(start, current_start)
-                            end = np.append(end, current_end)
-                        current_start = ptime
-                        current_end = None
-                        end_still_coming = True
-                        current_start_is_from_left = pl
+                        # Check if this opening time matches the previous closing time
+                        # If they match, continue the current interval instead of creating a new one
+                        if ptime == current_end:
+                            # Continue the current interval
+                            end_still_coming = True
+                            current_start_is_from_left = pl
+                        else:
+                            # we have an opening and a closing paranthesis
+                            if current_start != current_end:
+                                # we have a non-zero interval
+                                start = np.append(start, current_start)
+                                end = np.append(end, current_end)
+                            current_start = ptime
+                            current_end = None
+                            end_still_coming = True
+                            current_start_is_from_left = pl
```

Comment:
The following lines were the same irrespective of whether `ptime == current_end` or not:
```
end_still_coming = True
current_start_is_from_left = pl
```
So, how about:
```suggestion
                        # Check that this opening time doesn't match the previous closing time
                        # If they match, continue the current interval instead of creating a new one
                        if ptime != current_end:
                            # we have an opening and a closing paranthesis
                            if current_start != current_end:
                                # we have a non-zero interval
                                start = np.append(start, current_start)
                                end = np.append(end, current_end)
                            current_start = ptime
                            current_end = None
                        end_still_coming = True
                        current_start_is_from_left = pl
```